src/action/macos/create_apfs_volume.rs

```diff
@@ -145,14 +146,42 @@ impl Action for CreateApfsVolume {
             tracing::debug!("Volume was already unmounted, can skip unmounting")
         }
 
-        execute_command(
-            Command::new("/usr/sbin/diskutil")
-                .process_group(0)
-                .args(["apfs", "deleteVolume", &self.name])
-                .stdin(std::process::Stdio::null()),
-        )
-        .await
-        .map_err(Self::error)?;
+        // NOTE(cole-h): We believe that, because we're running the unmount force -> deleteVolume
+        // commands in an automated fashion, there's a race condition where we're running them too
+        // close to each other, so the OS doesn't notice the volume has been unmounted / hasn't
+        // completed its "unmount the volume" tasks by the time we try to delete it. If that is the
+        // case (unfortunately, we have been unable to reproduce this issue on the machines we have
+        // access to!), then trying to delete the volume 10 times -- with 500ms of time between
+        // attempts -- should alleviate this.
+        // https://github.com/DeterminateSystems/nix-installer/issues/1303
+        // https://github.com/DeterminateSystems/nix-installer/issues/1267
+        // https://github.com/DeterminateSystems/nix-installer/issues/1085
+        let mut retry_tokens: usize = 10;
+        loop {
+            let mut command = Command::new("/usr/sbin/diskutil");
+            command.process_group(0);
+            command.args(["apfs", "deleteVolume", &self.name]);
+            command.stdin(std::process::Stdio::null());
+            tracing::trace!(%retry_tokens, command = ?command.as_std(), "Waiting for volume deletion to succeed");
```

Comment:
Move this into the retry_tokens.saturating_sub(1) branch?

Reply:
(I argued against this because we are indeed waiting for it to succeed, whether or not we ever start retrying :P And also means I don't have to fix the other locations like `retry_bootstrap`, etc)